### ISSUE TITLE
Case insensitive comparison through Quality of Life method `String#=~(other : String)`

### ISSUE THREAD

**Issue by Sija** [crystal-lang/crystal#14667]
## Feature Request

I propose adding a new QoL method `String#=~(other : String)` with the above implementation:

```cr
class String
  def =~(other : String, *, options : Unicode::CaseOptions = :none) : Int32?
    compare(other, case_insensitive: true, options: options).tap do |comparison|
      return if comparison.zero?
    end
  end
end
```

This IMO supplements `String#=~(regex : Regex)`.

**Comment by straight-shoota** [comment]
A short cut for case-insensitive comparison could be handy.

But there are some issues with `=~`: It's not explicit about its behaviour, and that can lead to misinterpretation.

From `x =~ y`, knowing both arguments are strings, it's far from obvious what the operator would mean.
For example, deriving from the behaviour of `=~` with `Regex` (`"a b c" =~ /b/ # => 2`) could suggest semantics more similar to `String#includes?`.

`x.compare(y, case_insensitive: false)` is a lot more bulky, but that makes it explicit and thus it's very clear what it does.

**Comment by ysbaddaden** [comment]
I wish we had a case insensitive equality operator; `#compare(case_insensitive: true) == 0` is painful; yet `=~` already has too much meaning.

**Comment by Sija** [comment]
@ysbaddaden Yup, it was my wish as well, and the `=~` operator was the closest (and only) one available. `#compare` has its counterpart in a form of `<=>` operator, whilst no such exist for case insensitive comparisons unfortunately.

**Comment by ysbaddaden** [comment]
Maybe just `iequal?(other) : Bool`, `icompare`, ... but now that I write them, they're too far from being descriptive.
